pkg/utils/build/quay.go

```diff
@@ -103,3 +104,34 @@ func DoesTagExistsInQuay(imageURL string) (bool, error) {
 	}
 	return false, nil
 }
+
+func IsImageRepoPublic(quayImageRepoName string) (bool, error) {
+	isPublic, err := quayClient.IsRepositoryPublic(quayOrg, quayImageRepoName)
+	return isPublic, err
+}
+
+func DoesQuayOrgSupportPrivateRepo() (bool, error) {
+	repositoryRequest := quay.RepositoryRequest{
+		Namespace:   quayOrg,
+		Visibility:  "private",
+		Description: "Test private repository",
+		Repository:  constants.SamplePrivateRepoName,
+	}
+	repo, err := quayClient.CreateRepository(repositoryRequest)
+	if err != nil {
+		if err.Error() == "payment required" {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+	if repo == nil {
+		return false, fmt.Errorf("%v repository created is nil", repo)
+	}
+	// Delete the created image repo
+	_, err = DeleteImageRepo(constants.SamplePrivateRepoName)
+	if err != nil {
+		return true, fmt.Errorf("error while deleting private image repo: %v", err)
```

Comment:
Does it make sense to fail here if we know the answer for the private repository support?

Reply:
the reason I kept the error handling in case of delete image repo fails is that if  it start failing to delete the sample image repo, we will hit the limit of private repos and ultimately the test will start to fail. So I think its better to keep handling error here.